From 6c94dfe29aeacb9db00a3923d9829fe09581028c Mon Sep 17 00:00:00 2001
From: Vishal Sagar <vishal.sagar@xilinx.com>
Date: Mon, 2 Nov 2020 01:23:52 -0800
Subject: [PATCH 1741/1852] v4l: xilinx: sdirxss: Rate limit irq handler error
 messages

commit 62ea514294a0c9a80455e51f1f4de36e66e8c546 from
https://github.com/Xilinx/linux-xlnx.git

Sometimes in 3GA 1920x1080p60 mode the ST352 valid register is 0x0 even
though the core is locked. This causes the irq handler to print the
error messages leading to boot console flooding with messages.
These messages stop when the GT is reset. But to even do that console is
not available sometimes. So rate limit the error messages.

Signed-off-by: Vishal Sagar <vishal.sagar@xilinx.com>
Reported-by: Venkateshwar Rao Gannavarapu <venkateshwar.rao.gannavarapu@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/media/platform/xilinx/xilinx-sdirxss.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/media/platform/xilinx/xilinx-sdirxss.c b/drivers/media/platform/xilinx/xilinx-sdirxss.c
index f1afc561b4dd..097f513b4ea8 100644
--- a/drivers/media/platform/xilinx/xilinx-sdirxss.c
+++ b/drivers/media/platform/xilinx/xilinx-sdirxss.c
@@ -1096,7 +1096,7 @@ static int xsdirx_get_stream_properties(struct xsdirxss_state *state)
 	valid = xsdirxss_read(core, XSDIRX_ST352_VALID_REG);
 
 	if (mode >= XSDIRX_MODE_3G_MASK && !valid) {
-		dev_err(core->dev, "No valid ST352 payload present even for 3G mode and above\n");
+		dev_err_ratelimited(core->dev, "No valid ST352 payload present even for 3G mode and above\n");
 		return -EINVAL;
 	}
 
@@ -1527,7 +1527,7 @@ static irqreturn_t xsdirxss_irq_handler(int irq, void *dev_id)
 				state->vidlocked = true;
 				xsdirxss_set_gtclk(state);
 			} else {
-				dev_err(core->dev, "Unable to get stream properties!\n");
+				dev_err_ratelimited(core->dev, "Unable to get stream properties!\n");
 				state->vidlocked = false;
 			}
 
-- 
2.31.1

